include/saber/geometry/detail/simd_sse.hpp

```diff
@@ -0,0 +1,355 @@
+#ifndef SABER_GEOMETRY_DETAIL_SIMD_SSE_HPP
+#define SABER_GEOMETRY_DETAIL_SIMD_SSE_HPP
+#pragma once
+
+// sse
+#include <emmintrin.h>
+#include <immintrin.h>
+#include <xmmintrin.h>
```

Comment:
because each version of SSE is a superset of its earlier version...
i think only `<immintrin.h>` needed here.

since as a superset, it should already include `<emmintrin.h>` and `<xmmintrin.h>`.
check into that and make sure I'm correct!